data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java

```diff
@@ -541,6 +547,108 @@ private StateSyncCommitteesData getSyncCommitteesFromState(
             committeeIndices, spec.atEpoch(epoch).getConfig().getTargetCommitteeSize()));
   }
 
+  public SafeFuture<Optional<SyncCommitteeRewardData>> getSyncCommitteeRewardsFromBlockId(
+      final String blockId, final Set<String> validators) {
+    return getBlockAndMetaData(blockId)
+        .thenCompose(
+            result -> {
+              if (result.isEmpty() || result.get().getData().getBeaconBlock().isEmpty()) {
+                return SafeFuture.completedFuture(Optional.empty());
+              }
```

Comment:
to my reading, result.isEmpty() would end up being a 404?

Reply:
Yes, that's correct. Wouldn't that be expected?

Reply:
in what circumstance is `result.get().getData().getBeaconBlock().isEmpty()` true?